fastapi/dependencies/utils.py

```diff
@@ -454,77 +455,76 @@ async def solve_generator(
     return await stack.enter_async_context(cm)
 
 
+@dataclasses.dataclass
+class DependencySolverContext:
```

Comment:
Why are you using `dataclases.dataclass` here instead of Pydantic?

I just want to know if there's a reason in particular

Reply:
Dataclass is basically a quick way to declare a class with initializers.  I'm not looking for pydantic's validation.  This is supposed to be low overhead.
Could have used a plain class instead and written my own __init__().
Could also have made the solve_dependencies() an instance method, but I don't see much of that kind of programming style in FastAPI so I left it functional.

Reply:
Indeed, very good PR. I really liked it.